### PR TITLE
Auth final

### PR DESCRIPTION
**Pull request by finlaylp**
Simulator Screen Shot - iPhone 12 Pro Max - 2021-01-11 at 10 41 07
Simulator Screen Shot - iPhone 12 Pro Max - 2021-01-11 at 10 41 19
Added forgot password to the login page. It's kind of mostly handled by firebase

### CODE REVIEW

File: components/Authentication/login.jsx
Comment (jpoist97) [inline components/Authentication/login.jsx:90]: Do we need this to auto-login if the user has logged in before?

File: components/Authentication/signup.jsx
Comment (jpoist97) [inline components/Authentication/signup.jsx]: I don't quite understand this code, I didn't see your comment last night and idk where it went anymore haha.

**Review by jpoist97 — COMMENTED** [review]
Same general comments as last time, forgot password looks good! One small thing that would be nice is if you tapped off of the text the keyboard would hide, but not 100% necessary rn

File: components/Authentication/login.jsx
Comment (finlaylp) [reply components/Authentication/login.jsx:90]: no. that's done in app.js see line 136

File: components/Authentication/signup.jsx
Comment (finlaylp) [reply components/Authentication/signup.jsx]: All good. It is basically just there to stop it from triggering the first time, since useEffect triggers on init of a state var so I didn't want all the invalid field text things to show when the user hasn't even clicked on the input lines. So after they click on the input, it will allow it to show the invalid text thing.

**Comment by finlaylp** [comment]
> Same general comments as last time, forgot password looks good! One small thing that would be nice is if you tapped off of the text the keyboard would hide, but not 100% necessary rn

I'll try and add that rq

File: components/Authentication/login.jsx
Comment (jpoist97) [reply components/Authentication/login.jsx:90]: Does it still work? It doesn't do it for me anymore when I run it locally

Comment (finlaylp) [reply components/Authentication/login.jsx:90]: hmm. Well this is just a listener for auth state change, so if they log in or log out it triggers. So it shouldn't be what is automatically logging you in

File: components/Authentication/signup.jsx
Comment (jpoist97) [inline components/Authentication/signup.jsx]: nit: These can be simplified into one statement with a ternary operator.

File: components/Authentication/signup.jsx
Comment (jpoist97) [reply components/Authentication/signup.jsx]: Okay, that makes sense. I feel like there's probably an easier way to do this, but I can't think of one off the top of my head and I think its fine cause this works

**Review by jpoist97 — COMMENTED** [review]
Looks good to me, only concern is the auto-logging in stuff, otherwise I think this should be ready to go in soon.

File: components/Authentication/signup.jsx
Comment (finlaylp) [reply components/Authentication/signup.jsx]: I thought so too. This was the only way I could figure out, it was more of an issue than I expected

File: components/Authentication/signup.jsx
Comment (finlaylp) [reply components/Authentication/signup.jsx]: good catch

File: components/Authentication/login.jsx
Comment (jpoist97) [reply components/Authentication/login.jsx:90]: hm does it still work for you? Maybe it takes a second for Firebase to realize you're logged in so this was catching it previously?

Comment (finlaylp) [reply components/Authentication/login.jsx:90]: Not working for me either, although I'm not sure if it should be remembering you between simulator runs since it is a new package on expo. In theory App.js should be doing that part, after initializing firebase it checks for a user. By default firebase keeps users logged in so that should not be null if they are. But idk I'll look into it more before merging

Comment (jpoist97) [reply components/Authentication/login.jsx:90]: Okay, I think everything else in the PR is ready to merge so once we figure that out you can just re-request my review and I'll approve it

**Review by ctperry0301 — DISMISSED** [review]
In addition to what JP said, make sure that you dont leave any TODOs in this. Otherwise, this is really good. Good stuff

File: App.js
Comment (jpoist97) [inline App.js]: Is this used anywhere?

File: components/Authentication/login.jsx
Comment (jpoist97) [inline components/Authentication/login.jsx]: Do we still need this here?

**Review by jpoist97 — DISMISSED** [review]
Looks good, just comments that don't affect functionality

File: App.js
Comment (jpoist97) [reply App.js]: referring to the displayName variable

File: components/Authentication/login.jsx
Comment (finlaylp) [reply components/Authentication/login.jsx]: Nope accident good catch

File: App.js
Comment (finlaylp) [reply App.js]: well it will be used as the welcomeName once we update that from being 'Shriya'. But I can delete it since rn it's unnecessary

File: components/Authentication/signup.jsx
Comment (ctperry0301) [inline components/Authentication/signup.jsx:92]: if you put in else around each of these "actions", then you could combine the three useEffectes into one while still having the same functionality.

Comment (ctperry0301) [reply components/Authentication/signup.jsx:92]: In this case tho, I think it's better to have multiple useEffects like you have. Ignore the comment above.

**Review by jpoist97 — APPROVED** [review]
Looks ready for me, as long as Cole's cool with it I think this can go in

File: components/Authentication/signup.jsx
Comment (finlaylp) [reply components/Authentication/signup.jsx:92]: yeah cuz they're only triggering when the specific field they correlate with is changed, whereas if they were all in one they'd all trigger from a single of the textinputs

**Pull request merged**
